File: C++/Windows Forms Application/Windows Forms Application/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace Windows_Forms_Application
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {

        }

        private void label1_Click(object sender, EventArgs e)
        {

        }

        private void button1_Click(object sender, EventArgs e)
        {   
            // Taking the input into string and then later we will convert it to numeric
            string input = textBox1.Text;

            // Check if the input is not empty and show the error message
            double numericValue = 0;
            if (string.IsNullOrEmpty(input) || !double.TryParse(input, out numericValue))
            {
                label4.Text = "You must input a numeric value for the unit price";
                numericValue = 0;
                return;
            }

            if (comboBox2.SelectedItem == null)
            {
                label4.Text = "You must choose a quantity";
                return;
            }

            else if (comboBox1.SelectedIndex == -1)
            {
                label4.Text = "You must select a province";
                return;
            }

            // Calculate Combox1 selection into numeric value
            int quantity = int.Parse(comboBox2.SelectedItem.ToString());

            // Calculate the total value
            double total = numericValue * quantity;

            // Getting the selected province into string
            string selectedProvince = comboBox1.SelectedItem.ToString();

            // Calculate the tax rate based on province selection
            double taxRate;
            switch (selectedProvince)
            {
                case "Ontario":
                    taxRate = 0.13;
                    break;
                case "New Brunswick":
                case "Newfoundland and Labrador":
                case "Nova Scotia":
                case "Prince Edward Island":
                    taxRate = 0.15;
                    break;
                default:
                    taxRate = 0.05;
                    break;
            }

            // Calculate tax amount
            double taxAmount = total * taxRate;

            // Calculate the total with Tax
            double taxableAmount = total + taxAmount;

            // Displaying message at end after all the inputs
            label4.Text = "Subtotal before GST/HST: " + total + "\n" +
               "GST/HST: " + taxAmount + "\n" +
                "Total with GST/HST: " + taxableAmount;
            
        }

        private void label2_Click(object sender, EventArgs e)
        {

        }

        private void label3_Click(object sender, EventArgs e)
        {

        }

        private void textBox1_TextChanged(object sender, EventArgs e)
        {

        }

        private void comboBox2_SelectedIndexChanged(object sender, EventArgs e)
        {

        }

        private void comboBox1_SelectedIndexChanged(object sender, EventArgs e)
        {

        }

        private void label4_Click(object sender, EventArgs e)
        {
            
        }
    }
}

File: VB.NET/Windows-Forms-Application/WindowsApp/Form1.vb
Public Class Form1
    Private Sub TextBox1_TextChanged(sender As Object, e As EventArgs) Handles TextBox1.TextChanged

    End Sub

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        Dim input As String = TextBox1.Text

        ' Check if the input is not empty
        Dim numericValue As Double
        If String.IsNullOrEmpty(input) Or Not Double.TryParse(input, numericValue) Then
            MessageBox.Show("You must input a numeric value for the unit price")
            Return
        End If

        ' Check if ComboBox1 and ComboBox2 are set to valid values
        If ComboBox1.SelectedItem = 0 Then
            MessageBox.Show("You must choose a quantity")
            Return
        Else If ComboBox2.SelectedIndex = -1 Then
            MessageBox.Show("You must select a province")
            Exit Sub
        End If

       

        ' Calculate ComboBox1 selection into numeric value
        Dim quantity As Integer = Integer.Parse(ComboBox1.SelectedItem.ToString())

        ' Calculating the total value
        Dim total As Double = numericValue * quantity

        ' Getting the selected province into string
        Dim selectedProvince As String = ComboBox2.SelectedItem.ToString()

        ' Check ComboBox2 dropdown is selected
        If ComboBox2.SelectedItem IsNot Nothing Then
            selectedProvince = ComboBox2.SelectedItem.ToString()
        End If

        ' Calculate tax rate based on province selection
        Dim taxRate As Double
        Select Case selectedProvince
            Case "Ontario"
                taxRate = 0.13
            Case "New Brunswick", "Newfoundland and Labrador", "Nova Scotia", "Prince Edward Island"
                taxRate = 0.15
            Case Else
                taxRate = 0.05
        End Select

        ' Calculate tax amount
        Dim taxAmount As Double = total * taxRate

        ' Calculate the total with Tax
        Dim taxableAmount As Double = total + taxAmount

        ' Displaying message at end after all the inputs 
        MessageBox.Show("Subtotal before GST/HST: " & total.ToString("C") & vbCrlf &
                        "GST/HST: " & taxAmount.ToString("C") & vbCrlf &
                        "Total with GST/HST: " & taxableAmount.ToString("C")) 

    End Sub

    Private Sub Label1_Click(sender As Object, e As EventArgs) Handles Label1.Click

    End Sub

    Private Sub Label2_Click(sender As Object, e As EventArgs) Handles Label2.Click

    End Sub

    Private Sub ComboBox1_SelectedIndexChanged(sender As Object, e As EventArgs) Handles ComboBox1.SelectedIndexChanged

    End Sub

    Private Sub ComboBox2_SelectedIndexChanged(sender As Object, e As EventArgs) Handles ComboBox2.SelectedIndexChanged

    End Sub
End Class
